test/browser-tests/window-utils/serviceworkers/immediate-control.js

```diff
@@ -0,0 +1,2 @@
+self.addEventListener('install', () => self.skipWaiting());
```

Comment:
Could you add a comment here just along the lines of:

"This service worker does not use event.waitUntil() on purpose. This has ambiguity over implementation across browsers (i.e. whether it's needed to be used with skipWaiting and clients.claim()). This service worker is used to test controlledBySw() - PLEASE DO NOT CHANGE."

Without this I can see someone eventually sneaking in event.waitUntil() (i.e. myself) and then it's undermines the purpose of this new method.